sections/src/StudentSectionCard.js

```diff
@@ -38,63 +42,97 @@ export default function StudentSectionCard({
-            action={!enrolledInThisSection}
-            onClick={() => joinSection({ target_section_id: section.id })}
-          >
-            {enrolledInThisSection ? (
-              <div>Switch to Section {slotText}</div>
+    <>
+      <Card
+        border={enrolledInThisSection || teachingThisSection ? "primary" : null}
+      >
+        <Card.Body>
+          <Card.Title>
+            {isStaff &&
+              (section.staff == null
+                ? config.canTutorsChange && (
+                    <Button
+                      className="float-right"
+                      size="sm"
+                      onClick={() => claimSection({ section_id: section.id })}
+                    >
+                      Claim
+                    </Button>
+                  )
+                : (section.staff.email === currentUser?.email
+                    ? config.canTutorsChange
+                    : config.canTutorsReassign) && (
+                    <Button
+                      className="float-right"
+                      size="sm"
+                      variant="danger"
+                      onClick={() =>
+                        unassignSection({ section_id: section.id })
+                      }
+                    >
+                      Unassign
+                    </Button>
+                  ))}
+            {!isStaff && <Tags tags={section.tags} />}
+            {isStaff ? (
+              <Link to={`/section/${section.id}`}>{title}</Link>
             ) : (
-              <span className="btn-link">
-                {enrolledSection == null ? "Join Section" : "Switch to Section"}{" "}
-                {slotText}
-              </span>
+              title
             )}
-          </ListGroup.Item>
-        ) : null}
-      </ListGroup>
-    </Card>
+          </Card.Title>
+          <Card.Text>{section.description}</Card.Text>
+        </Card.Body>
+        <ListGroup variant="flush">
+          {section.students.map((student, i) => (
+            <ListGroup.Item
+              key={i}
+              active={student.email === currentUser?.email}
+            >
+              {enrolledInThisSection || isStaff ? student.name : "A student"}
+            </ListGroup.Item>
+          ))}
+          {hasSpace && !isStaff && config.canStudentsChange ? (
+            <ListGroup.Item
+              disabled={enrolledInThisSection}
+              action={!enrolledInThisSection}
+              onClick={joinSectionWorkflow}
+            >
+              {enrolledInThisSection ? (
+                <div>Switch to Section {slotText}</div>
+              ) : (
+                <span className="btn-link">
+                  {enrolledSection == null
+                    ? "Join Section"
+                    : "Switch to Section"}{" "}
+                  {slotText}
+                </span>
+              )}
+            </ListGroup.Item>
+          ) : null}
+        </ListGroup>
+      </Card>
+      <EnterEnrollmentCodeModal
```

Comment:
not sure why the diff is messed up, but in this file I just added the EnterEnrollmentCodeModal and changed the join section stuff to use the functions defined at the top of the file. (`onClick={joinSectionWorkflow}`)

Reply:
(note) the diff here is that everything is wrapped in an extra pair of `<></>`, since there are now two elements in the return expression instead of one